### PR TITLE
Working close tab button

### PR DESCRIPTION
**Pull request by PiotrSquaresMoscicki**

### CODE REVIEW

File: PolyDock/PolyDock/src/pd/ecs/cmp/tabsHeader/TabsHeaderWidget.cpp [hunk 1]
```diff
@@ -226,6 +250,20 @@ AlignedBox2i DefaultTabsHeaderWidget::getTextRectAtIdx(int idx) const
...
 }
 
+// ---------------------------------------------------------------------------------------------------------
+AlignedBox2i DefaultTabsHeaderWidget::getCloseButtonRectAtIdx(int idx) const
+{
+	const AlignedBox2i tabRect = getTabRectAtIdx(idx);
+	const int top = tabRect.min().y() + 8;
+	const int right = tabRect.max().x() - 8;
+	const int bottom = tabRect.max().y() - 8;
+	const int left = right - (bottom - top);
```
Comment (JDuchniewicz) [inline PolyDock/PolyDock/src/pd/ecs/cmp/tabsHeader/TabsHeaderWidget.cpp:260]: this is cryptic :/ Either math is weird or I it is some shortcut, which can baffle a developer in future

File: PolyDock/PolyDock/src/pd/ecs/cmp/tabsHeader/TabsHeaderWidget.hpp [hunk 2]
```diff
@@ -17,14 +17,26 @@ namespace pd::ecs::cmp::tabsHeader
 	public:
 		enum class eAddButtonState { IDLE, HOVERED, PRESSED };
 
+		class UpdateParams
```
Comment (JDuchniewicz) [inline PolyDock/PolyDock/src/pd/ecs/cmp/tabsHeader/TabsHeaderWidget.hpp:20]: This I very much like :+1:

File: PolyDock/PolyDock/src/pd/ecs/sys/tabsHeader/TabsClosing.cpp [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+namespace pd::ecs::sys
+{
+
+// ---------------------------------------------------------------------------------------------------------
+void CloseButtonHoverSystem::update(entt::registry& registry, entt::entity root) const
+{
+	if (const auto* inputComponent = registry.try_get<root::Input>(root))
+	{
+		auto view = registry.view<tabsHeader::Widget, tabsHeader::Component>();
```
Comment (JDuchniewicz) [inline PolyDock/PolyDock/src/pd/ecs/sys/tabsHeader/TabsClosing.cpp:20]: let's decide on one syntax for this view iteration, either horizontally or vertically

File: PolyDock/PolyDock/src/pd/ecs/sys/tabsHeader/TabsRemoving.cpp [hunk 4]
```diff
@@ -23,23 +23,28 @@ void TabsRemovalSystem::update(entt::registry& registry, entt::entity root) cons
...
-                else
-                    activeCmp.activeTab = tabsHeader.tabs().at(activeIdx - 1);
-            }
-
-            tabsHeader.closeTab(requestCmp.tabToRemove);
-            registry.destroy(requestCmp.tabToRemove);
-
-            registry.get_or_assign<tabsHeader::WidgetUpdateRequest>(entity);
-            registry.get_or_assign<tabsHeader::SelectedTabs>(entity).selectedTabs = { activeCmp.activeTab };
+			for (const entt::entity& toRemove : requestCmp.tabsToRemove)
```
Comment (JDuchniewicz) [inline PolyDock/PolyDock/src/pd/ecs/sys/tabsHeader/TabsRemoving.cpp:26]: either it is GH rendering issue, or indentation is broken

File: PolyDock/PolyDock/src/pd/ecs/sys/tabsHeader/TabsClosing.cpp (hunk 3 above)
Comment (PiotrSquaresMoscicki) [reply PolyDock/PolyDock/src/pd/ecs/sys/tabsHeader/TabsClosing.cpp:20]: What do you mean?

**Pull request closed without merging**